File: project/backend-services/smart-irrigation-backend/infrastructure/endpoint/graphql/src/main/java/pt/sensae/services/smart/irrigation/backend/infrastructure/endpoint/graphql/model/ParkSensorDataHistoryDetails.java
package pt.sensae.services.smart.irrigation.backend.infrastructure.endpoint.graphql.model;

public record ParkSensorDataHistoryDetails(String id,
                                           Long reportedAt,
                                           IlluminanceDataDetails illuminance,
                                           SoilMoistureDataDetails soilMoisture) implements SensorDataHistoryDetails {
}

File: project/frontend-services/libs/smart-irrigation/model/src/lib/model/history/ParkSensorDataHistoryDetails.ts
import {SensorDataHistoryDetails} from "./SensorDataHistoryDetails";
import {SoilMoistureDataDetails} from "../details/SoilMoistureDataDetails";
import {IlluminanceDataDetails} from "../details/IlluminanceDataDetails";

export class ParkSensorDataHistoryDetails extends SensorDataHistoryDetails {
  constructor(id: string, reportedAt: Date, public soilMoisture: SoilMoistureDataDetails, public illuminance: IlluminanceDataDetails) {
    super(id, reportedAt)
  }
}
